.github/workflows/deploy.yml

```diff
@@ -1,88 +1,87 @@
 name: deploy
 
-on: 
+on:
   push:
     branches:
       - main
-  release:
-      types: [published]
+    tags:
+      - 'v*'
+
+env:
+  # We don't want to do certain steps if this is running in a fork
+  IS_ORIGINAL_REPO: ${{ github.repository == 'microbiomedata/nmdc-server' }}
+
+  # Used to determine if we add the special `prod` Docker image tag
+  IS_PROD_RELEASE: ${{ startsWith(github.ref, 'refs/tags/v') }}
+
+  # Used when sending redeploy action requests to Rancher
+  RANCHER_NAMESPACE: ${{ startsWith(github.ref, 'refs/tags/v') && 'nmdc' || 'nmdc-dev' }}
 
 jobs:
   build:
     runs-on: ubuntu-latest
+
     strategy:
       matrix:
         image: [server, client, worker]
 
     steps:
-      -
-        name: Checkout code
-        uses: actions/checkout@v2
+      - name: Checkout code
+        uses: actions/checkout@v4
         with:
-          submodules: 'recursive'
-      -
-        name: Login to GitHub Container Registry
-        uses: docker/login-action@v1
+          # history for all branches and tags is needed for setuptools-scm
+          fetch-depth: 0
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
```

Comment:
What are these actions giving us? It doesn't look like the previous version of this action used these steps.